src/components/datagrid/data_grid.tsx

```diff
@@ -320,36 +322,34 @@ function createKeyDownHandler(
     } else if (keyCode === keyCodes.PAGE_DOWN) {
       if (props.pagination) {
         event.preventDefault();
-        const totalRowCount = props.rowCount;
+        const rowCount = props.rowCount;
         const pageIndex = props.pagination.pageIndex;
         const pageSize = props.pagination.pageSize;
-        const pageCount = Math.ceil(totalRowCount / pageSize);
-        if (pageIndex < pageCount) {
-          props.pagination!.pageIndex = pageIndex + 1;
-          props.pagination.onChangePage(props.pagination.pageIndex);
-          const newPageRowCount = computeVisibleRows(props);
+        const pageCount = Math.ceil(rowCount / pageSize);
+        if (pageIndex < pageCount - 1) {
+          props.pagination.onChangePage(pageIndex + 1);
+          const newPageRowCount = computeVisibleRows({
```

Comment:
I refactored `computeVisibleRows` to only take specifically `rowCount` and `pagination`, allowing those to be more customized when passing into that function. This was necessary due to no longer directly modifying `props`